site/public/js/userform.js

```diff
@@ -51,7 +51,6 @@ function newUserForm() {
     $("#edit-student-modal-title").css('display','none');
     $("#edit-grader-modal-title").css('display','none');
     $("#user-form-already-exists-error-message").css('display','none');
-    $('[name="edit_user"]', form).val("false");
```

Comment:
I do not think you want to remove this line. With your code, if you edit a user, press cancel, then add a new user. The edit_user value will be true. 
Lines 208/209 might also relate to this issue.